### DISCUSSION TITLE
Don't require a unique name

### DISCUSSION THREAD

**Discussion by Rolanddoda** [vuejs/pinia#989] [Ideas]
The docs say:

> a store is defined using defineStore() and it requires a **unique** name, passed as the first argument. This name also referred to as id, is necessary and is used by Pinia to connect the store to the dev-tools.

Is there any way we can omit the id? I think it only adds more complexity to defining a store because you have to maintain all the names of the stores since they also need to be unique, but also you have to come up with names.

I understand that the name is needed for Devtools, but is there any way to automatically infer it? For example, when creating Vue components, you can add a name property and is what will be shown in Devtools. But if you don't specify a name, Devtools will automatically infer the name of the component.

**Comment by posva** [comment]
Names are necessary to ensure the state always contains the same keys in server and client code

**Reply by Rolanddoda** [reply]
I see. Does that mean there is no way to make them optional and if one it not provided to generate one by default ?

**Reply by posva** [reply]
there could be a vite plugin that generates the ids based on the filepath but I don't think there is another way to generate a consistent unique id that doesn't involve the build process

**Comment by tbontb-iaq** [comment]
I have an idea: wrap pinia's `defineStore` with a custom `defineStore` and fill in the name automatically.

One possible method is to obtain the file name, line number and column number through the stack information of `new Error().stack`.
